### PR TITLE
MOTR-56 Download page

### PR DESCRIPTION
**Pull request by akre96**
**Primary Changes:**

1. Creates a page for the downloading of data (No actual download functionality yet)
2. Filterable files in a list of cards format
3. Additional tests created for download reducer and page. Storybook stories also created
4. Icons created for animal vs human, custom icon font generated
5. Pagination of downloads implemented
6. "Add to cart" feature added
7. Branched off of Dashboard, so includes ChartJS graphs on dashboard and aggregate data figures.

### CODE REVIEW

File: src/components/allUploadStats.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+AllUploadStats.propTypes = {
+  allUploadStats: PropTypes.shape({
+    totalSamples: PropTypes.number.isRequired,
+    totalSize: PropTypes.number.isRequired,
+  }),
+};
+AllUploadStats.defaultProps = {
+  allUploadStats: stats,
```
[the remark this answers is not in the document]
Comment (jimmyzhen) [reply src/components/allUploadStats.jsx]: More mock async calls will be great!

In my experience in working on the ClinGen projects, the use of "loading states" for numerous async requests could become hard to maintain over time. If we can dispatch actions to update the store with the returned response, would it be enough to trigger the UI re-rendering?

Tackling this implementation in a separate branch after merging is a great idea! Thanks!

File: src/App.jsx [hunk 2]
```diff
@@ -14,16 +15,19 @@ import UploadScreenConnected from './components/uploadScreen';
...
+  applyMiddleware(thunkMiddleWare),
+);
 function App({ history = hist }) {
   // TODO: Before production remove redux devtools extension javascript
   return (
     <Provider store={createStore(rootReducer,
       defaultRootState,
-      window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__())
-      }
+      enchancer)}
```
[the remark this answers is not in the document]
Comment (akre96) [reply src/App.jsx]: Implemented change

File: src/assets/assayList.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  'TCA Cycle',
+  'Targeted Ceramides',
+  'Global Proteomics',
+  'Global Phospho-proteomics',
+  'Redox proteome',
+  'Acyl/Acetyl proteome',
+  'SomaLogic',
+];
+
+export default assayList.sort();
```
[the remark this answers is not in the document]
Comment (akre96) [reply src/assets/assayList.js]: Implemented change

File: src/assets/colors.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    dgray: '#404040',
+    lgray: '#8c8c8c',
+    dblue: '#183a7a',
+    lblue: '#8b9cbc',
+    dgreen: '#2d5e2d',
+    lgreen: '#96ae96',
+  },
+};
+
+export default colors;
```
[the remark this answers is not in the document]
Comment (akre96) [reply src/assets/colors.js]: Implemented change

File: src/assets/downloadFilters.jsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    icons: [(<span className="oi oi-circle-check green" />), (<span className="oi oi-loop-square dblue" />), (<span className="oi oi-ellipses dgray" />)],
+  },
+  {
+    keyName: 'type',
+    name: 'Assay Type',
+    filters: assayList,
+  },
+];
+
+export default downloadFilters;
```
[the remark this answers is not in the document]
Comment (akre96) [reply src/assets/downloadFilters.jsx]: Implemented change, created src/lib forlder

File: src/components/allUploadStats.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (akre96) [reply src/components/allUploadStats.jsx]: Will be done in future branch! (Creating loading visuals and updating UI through redux mock async function dispatches)

File: src/App.jsx [hunk 6]
```diff
@@ -14,16 +15,19 @@ import UploadScreenConnected from './components/uploadScreen';
 import AuthLoadingConnected from './components/authLoading';
 import LinkoutPageConnected from './components/linkoutPage';
 import AnalysisHomePageConnected from './components/analysisHomePage';
+import DownloadPageConnected from './components/downloadPage';
 
 const hist = History;
-
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+const enchancer = composeEnhancers(
```
[the remark this answers is not in the document]
Comment (akre96) [reply src/App.jsx]: Spelling has been fixed!

**Comment by jimmyzhen** [comment]
@akre96, thank you for the quick turnaround! Strong work!

**Comment by kilodalton** [comment]
@akre96 we will move ahead with the ticket as we have discussed but please make a ticket to resolve the console error

image

**Comment by kilodalton** [comment]
Merged MOTR-56 Download page

**Pull request merged**